### PR TITLE
Remove Spec.Strategy and rollback behavior

### PR DESCRIPTION
**Pull request by kalantar**
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>
Addresses https://github.com/iter8-tools/iter8/issues/969
- Moves Strategy.Actions to Spec
- Deletes Strategy.DeploymentPattern
- Deletes Strategy.Weights
- Deletes Objective.RollbackOnFailure

### CODE REVIEW

File: api/v2beta1/experiment_types.go [hunk 1]
```diff
@@ -60,8 +60,10 @@ type ExperimentSpec struct {
 	// +kubebuilder:validation:MinItems:=1
 	Versions []string `json:"versions" yaml:"versions"`
```
Comment (sriumcp) [inline api/v2beta1/experiment_types.go:61]: Could we actually retain `versionInfo` as the name of this field as opposed to `versions`? The idea is that `versionInfo` fields can appear wherever the user is expected to supply information about versions (including experiment spec, which in this case happens to be a simple list of identifiers).

**Review by sriumcp — COMMENTED** [review]
minor change requested

File: api/v2beta1/experiment_types.go (hunk 1 above)
Comment (kalantar) [reply api/v2beta1/experiment_types.go:61]: @sriumcp This is planned as part of issue https://github.com/iter8-tools/iter8/issues/963 (that part still to be done). This name is temporary until I've had a chance to go through all uses of "versionInfo" in the controller (includimg documentation). It is an issue that can now be addressed.  I'll do it after the current issue I've already started.

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**